src/dbnode/storage/types.go

```diff
@@ -466,6 +466,10 @@ type databaseShard interface {
 		bootstrappedSeries *result.Map,
 	) error
 
+	// Load does the same thing as Bootstrap, except it can be called more than once
+	// and after a shard is bootstrapped already.
+	Load(series *result.Map) error
```

Comment:
Does it not make sense perhaps to just allow bootstrap to be called multiple times and that bootstrapping a bootstrapped shard is ok?

Reply:
Its more complicated than that because if a shard is marked `bootstrapping` it can't serve reads and there may be some other implications I hadn't considered also. I think this strikes the best balance between sharing as much code as possible while not interfering with existing logic/workloads. I also think allowing `Bootstrap` to be called more than once is confusing as long as there is a difference between a bootstrapped and unbootstrapped shard

Reply:
Yeah, I rethought this, makes sense more to just make calling Load time and time again and then Bootstrap once.